nav2_behavior_tree/include/nav2_behavior_tree/plugins/condition/would_a_controller_recovery_help_condition.hpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2022 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__WOULD_A_CONTROLLER_RECOVERY_HELP_CONDITION_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__WOULD_A_CONTROLLER_RECOVERY_HELP_CONDITION_HPP_
+
+#include <string>
+
+#include "nav2_msgs/action/follow_path.hpp"
+#include "nav2_behavior_tree/plugins/condition/is_error_code_active_condition.hpp"
+
+namespace nav2_behavior_tree
+{
+
+class WouldAControllerRecoveryHelp : public IsErrorCodeActive
+{
+  using Action = nav2_msgs::action::FollowPath;
+  using ActionGoal = Action::Goal;
+
+public:
+  WouldAControllerRecoveryHelp(
```

Comment:
For the input port of `error_codes_to_check`, is there a way to override that here and provide the default of the BT controller blackboard ID for the error code? That way by default they don't need to think much about it?

I'm not 100% sure if that's possible or not, but would be a quality of life thing

Reply:
Actually in parallel to that, did we provide defaults for the error ID code fields for the Action BT nodes?

Reply:
I completely agree that it would be nicer if we could do that.

Reply:
Can't we have a `to_string` like we have a `convertFromString`?

Reply:
It seems that we are not able to override std::to_string. https://stackoverflow.com/questions/33399594/making-a-user-defined-class-stdto-string-able. :(.

Reply:
OK, that stinks, but not mission critical. I suppose we could hard-code it in the BT node constructors instead before getting the input port value (and if no value is set or is empty string, then use that)? But maybe we should just leave this alone.